extensions/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/BuildProcessor.java

```diff
@@ -128,9 +130,40 @@ void coreServiceFilter(BuildProducer<CamelServiceFilterBuildItem> filterBuildIte
         }
 
         @BuildStep
-        void serviceInfoTransformers(BuildProducer<CamelServiceInfoTransformerBuildItem> mapperBuildItems) {
-            mapperBuildItems.produce(
-                    new CamelServiceInfoTransformerBuildItem(CamelServiceInfoTransformers::configurer));
+        void coreServices(
+                ApplicationArchivesBuildItem archives,
+                BuildProducer<CamelServiceBuildItem> services) {
+
+            /* Registered only */
+            CamelSupport.propertyFiles(
+                    archives,
+                    "META-INF/services/org/apache/camel/component",
+                    "META-INF/services/org/apache/camel/language",
+                    "META-INF/services/org/apache/camel/dataformat")
```

Comment:
we need to fine tune language and data-format as we can store into the registry only singleton services (this requires also some changes to the FastCamelContext as it assumes languages and dataformats are stored in the registry)

Reply:
Yes we do have this problem as today but this is a good change to fix it

Reply:
> we need to fine tune language and data-format 

Not sure how should we tune?

> as we can store into the registry only singleton services 

You seem to imply language and data-format are not singleton services, but could you plz. define "singleton services"?

Reply:
The "tuning" here is about to carefully select which language/dataformat can be bound to the registry.

As example camel creates components only once so every time you use a schema, then you'll end up using the same component instance. 

For languages and dataformat that's not always true and this is because you can use the same dataformat multiple time in the same route with a different configuration.